dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/resources/de/digitalcollections/cudami/server/backend/impl/database/migration/V14.05.00__DDL_drop_tags_and_recreate.sql

```diff
@@ -0,0 +1,9 @@
+DROP TABLE IF EXISTS tags;
```

Comment:
Das dürfte Probleme geben wegen der Trigger aus V13.04.01, die die Tabelle `tags` referenzieren.

In dem Fall müßte man die nicht mehr benötigten Spalten `label`, `namespace`, `id` und `type` löschen und nur die Spalce `value` hinzufügen.

Außerdem muß der Unique-Index auf `(type, namespace,id`) noch entfernt werden.

Reply:
in der doku steht: "DROP TABLE always removes any indexes, rules, triggers, and constraints that exist for the target table. However, to drop a table that is referenced by a view or a foreign-key constraint of another table, CASCADE must be specified. (CASCADE will remove a dependent view entirely, but in the foreign-key case it will only remove the foreign-key constraint, not the other table entirely.)"

https://www.postgresql.org/docs/current/sql-droptable.html

Reply:
soll ich die trigger also wieder neu hinzufügen?

Reply:
die trigger sind nur bei wemi-objekten.... sollte der trigger nicht bei identifiable eingerichtet werden oder für alle identifiables extra?

Reply:
Das Drop Table oben verstehe ich so, dass nur die Trigger gelöscht werden, die auf dieser Tabelle (`CREATE TRIGGER ... ON tags`) angelegt werden, gelöscht werden, aber nicht diejenigen aus V13.04.01, die auf anderen Tabellen angelegt werden und nur zur Laufzeit beim Insert in diese anderen Tabellen aktiv werden. Möglicherweise könnten die entgegen meiner ersten Annahme doch das (kurzzeitige) Löschen der `tags`Tabelle überleben, müßte man prüfen.

Die Trigger müssen bei allen Identifiables eingerichtet werden, ich habe das entsprechende (allgemein gehaltene, da es nicht nur die Tags betrifft) Ticket eingestellt.

Aber am einfachsten wäre es wirklich, nur die nicht mehr benötigten Spalten der `tags` zu löschen, und die eine neue Spalte einzufügen.

Reply:
ok, done